src/main/java/pl/tomaszdziurko/jvm_bloggers/blog_posts/BlogPostsFetchingScheduler.java

```diff
@@ -6,21 +6,34 @@
 import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Component;
 import pl.tomaszdziurko.jvm_bloggers.TimeConstants;
+import pl.tomaszdziurko.jvm_bloggers.metadata.Metadata;
+import pl.tomaszdziurko.jvm_bloggers.metadata.MetadataKeys;
+import pl.tomaszdziurko.jvm_bloggers.metadata.MetadataRepository;
+import pl.tomaszdziurko.jvm_bloggers.utils.NowProvider;
 
 @Component
 @Slf4j
 public class BlogPostsFetchingScheduler {
```

Comment:
You can remove hand-crafted constructor and replace it with `@RequiredArgsConstructor(onConstructor = @__(@Autowired))` class level annotation.